core/src/robot/AbstractTransmission.cc

```diff
@@ -106,11 +116,13 @@ namespace jiminy
 
         // Clear the references to the robot
         robot_.reset();
+        notifyRobot_ = nullptr;
 
         // Unset the Id
         transmissionIdx_ = -1;
 
-        // TODO Delete motor and joint references
+        // Delete reference to motors
+        motors_.clear();
```

Comment:
There is more to clear. You should reset everything. If it is not the case for sensors and motors you can fix it.

Reply:
just to double check, you really mean everything? 
```
baseTransmissionOptions_(nullptr),
    transmissionOptionsHolder_(),
    isInitialized_(false),
    isAttached_(false),
    robot_(),
    notifyRobot_(),
    name_(name),
    transmissionIdx_(-1),
    jointNames_(),
    jointModelIndices_(),
    jointTypes_(),
    jointPositionIndices_(),
    jointVelocityIndices_(),
    motorNames_(),
    motorIndices_(),
    motors_(),
    forwardJacobian_(),
    backwardJacobian_()
```

Reply:
Maybe not everything. But everything public "constant" at least:
```
    isInitialized_
    isAttached_
    robot_
    transmissionIdx_
    jointNames_
    jointModelIndices_
    jointTypes_
    jointPositionIndices_
    jointVelocityIndices_
    motorNames_
    motorIndices_
```

Reply:
ok.. I do not want to annoy you but I dont really understand how you choose this. Why for you these are public "constant" ? they are all protected members, and what is different about the Jacobians ? Bc the jacobians can change but not the names , indices ect ? thank you I just try to understand the purpose of this

Reply:
They are never changing, except at initialization. Which is not the case for the Jacobian. So the user is already excepting the value of the jacobian being outdated most of the time, but not the fields that are constant except at initialization.

They seem to be protected but actually they are not really, since there are getters to access them. What I mean by public is everything the user can access. So providing getters for private/protected fields in the same has making the attributes read-only public (which does not exist in C++, so getters is the way to go).

Reply:
thanks a lot for the explicaiton, it makes sense to me